### PR TITLE
[WIP] [DO NOT MERGE] - Composer Based Setup and Improved Horde_Core_Controller routing

### PR DESCRIPTION
**Pull request by ralflang**
To be used in conjunction with a PR in Horde Base.

- Fix various scenarios for Horde_Core_Controller where $APP does not live below horde base app
- Add capabilities for Horde Controller based pages to have unauthenticated access or http basic auth

This has seen some closed-shop testing in the last few months and proved stable.

Do not merge unless you accept the changed content of composer.json

### CODE REVIEW

File: lib/Horde/Core/Db/Migration.php [hunk 1]
```diff
@@ -65,20 +65,59 @@ public function __construct($basedir = null, $pearconf = null)
         error_reporting($old_error_reporting & ~E_DEPRECATED);
         $pear = new PEAR_Config($pearconf);
 
+        // Detect and handle the Composer use case
+        if (class_exists('Composer\Autoload\ClassLoader')) {
```
Comment (ralflang) [inline lib/Horde/Core/Db/Migration.php]: Add second parameter to prevent accidental loading - it should already be loaded for this scenario

**Comment by bytestream** [comment]
@mrubinsk @yunosh @ralflang is there anything stopping this from progressing? Pear support has been removed in Composer 2 which is currently in alpha - https://github.com/composer/composer/issues/8778

**Comment by ralflang** [comment]
I am currently lacking time.

**Comment by ralflang** [comment]
The fork needs updating and I need to make the PR from a version which does not have the downstream composer.json

**Comment by ralflang** [comment]
Superseded by https://github.com/horde/Core/pull/10 - I am closing this one as it cannot get merged due to composer.json changes.

**Pull request closed without merging**